src/backend/components/py.pi/caliopen_pi/qualifiers/mail.py

```diff
@@ -44,7 +48,20 @@ def __init__(self, user):
 
     def _get_tags(self, message):
         """Evaluate user rules to get all tags for a mail."""
-        return []
+        tags = []
+        if message.privacy_features.get('is_internal', False):
+            # XXX do not hardcode the wanted tag
+            internal_tag = [x for x in self.user.tags if x.name == 'CALIOPEN']
+            if internal_tag:
+                # XXX need to transform UserTag to ResourceTag
+                tag = ResourceTag()
+                tag.date_insert = datetime.utcnow()
```

Comment:
I'm not on the same wavelength… `date_insert` is the date the tag was created/inserted in db, not the date it was nested into the object.
Furthermore, we'll change the model: `tags` in messages, contacts, etc. will be an array of IDs, not an object.

Reply:
Ok I'll wait for your update on nested tag structure to go further.